### PR TITLE
Fix data validation responseselector

### PR DESCRIPTION
**Pull request by m-vdb**
**Proposed changes**:
- Fixes https://github.com/RasaHQ/rasa/issues/6728
- TDD ✅ 

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by m-vdb** [comment]
@degiz or @federicotdn would you have time to review this today? It's needed for 2.0 release 🙏

File: data/test_selectors/nlu.yml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    - text: that's not something I can help with
+
+  utter_faq/placeholder:
+    - text: so
+
+  utter_faq/takeacut:
+    - text: No, my creators don't get a cut from the offsets you buy. I'm just here to help!
+
+
+stories:
```
Comment (degiz) [inline data/test_selectors/nlu.yml:243]: Do we need to keep the story inside `nlu.yaml` file, or we can move it into `data/test_selectors/stories.yml`?
I'd personally prefer not to have a story in a file called `nlu.yml`

File: tests/test_validator.py [hunk 2]
```diff
@@ -35,6 +35,18 @@ async def test_verify_valid_utterances():
     assert validator.verify_utterances()
 
 
+async def test_verify_valid_responses():
+    importer = RasaFileImporter(
+        domain_path="data/test_domains/selectors.yml",
+        training_data_paths=[
+            "data/test_selectors/nlu.yml",
+            "data/test_selectors/stories.yml",
```
Comment (degiz) [inline tests/test_validator.py:43]: Based on the previous comment, we might remove the `data/test_selectors/stories.yml` from here.

**Review by degiz — APPROVED** [review]
Thanks a lot for fixing that! 🚀 
Lest small comments, feel free to merge.

File: rasa/validator.py [hunk 3]
```diff
@@ -130,7 +130,12 @@ def verify_intents_in_stories(self, ignore_warnings: bool = True) -> bool:
 
     def _gather_utterance_actions(self) -> Set[Text]:
         """Return all utterances which are actions."""
-        return {
+
+        responses = {
+            response.split(RESPONSE_IDENTIFIER_DELIMITER)[0]
```
Comment (degiz) [inline rasa/validator.py:135]: Also, we decided to use constants with the full module path, not export them directly.

**Comment by m-vdb** [comment]
@degiz I've moved the stories in `stories.yml` but it doesn't seem that `RasaYamlReader` is ready for this change because it [only accepts yaml files that have the `responses` or `nlu` key](https://github.com/RasaHQ/rasa/blob/5005e98049b3a7d0e12c9f72d596d90dff1c50d6/rasa/shared/nlu/training_data/formats/rasa_yaml.py#L351), not the `stories` key. Making a change here would be a bit out of the scope of this PR, so I'll revert the changes I've made. Are we/you aware of this, though?

**Pull request merged**